Feature/image service

KianMiridoozini opened the pull request:
ImageService:

Using Imagick (php) to handle resizing and format convertion of uploaded images.

All uploaded images will be converted to webp format.

The width and hight get defined in the controllers.

Errors for illegal file-sizes and formats displayed on client side are displayed using a **centralized image-handler.js**

In addition some housekeeping has been made in terms of dependency usage and handling.

* This branch has some pull from style/improvements branch. So there are additional changes which are not related to the ImageService feature. Please Love me <3

KianMiridoozini committed:
Merge branch 'style/improvements' of https://github.com/jayb2302/Drinx into develop

KianMiridoozini committed:
Cocktail images

KianMiridoozini committed:
Profile Images

jayb2302 commented on public/assets/js/modules/cocktail.js:
```diff
@@ -89,4 +89,61 @@ export function initializeCocktail() {
         const units = JSON.parse(document.getElementById('unitOptions')?.value || '[]');
         return units.map(unit => `<option value="${unit.id}">${unit.name}</option>`).join('');
     }
+
```
It looks like this could be more organized some how and, Can you explain why we need so much js for both user and cocktail images ?

KianMiridoozini replied:
The JS is there to handle errors for larger images.

The images selected by the user, can not go through any processing before they click on Save change or submit, therefor php handling of the errors  will not work if we want to display the error inside the forms.

The JS is there to handle those errors, and to prevent image submission on the fly.

KianMiridoozini replied:
FIXED

jayb2302 commented on public/assets/js/modules/profile.js:
```diff
@@ -13,4 +13,53 @@ export function initializeProfile() {
             section.style.display = section.style.display === 'none' ? 'block' : 'none';
         }
     });
+   // Handle Image Preview and Validation for Profile Picture
```
I think it would be more optimal to have a helper module for example  imageHandler.js
that would combine the shared logic for handling image validation and previews so it will be a reusable logic avoiding duplicating same logic in multible modules.

KianMiridoozini replied:
Yes, this could work, but it would also require shared class name or Id in all form where image upload takes place

KianMiridoozini replied:
FIXED

KianMiridoozini committed:
ImageService:

Centralized Image-Handler for validating image size and format on the client side into "image-handler.js" module which is imported and initialized through main.js on selected pages.

jayb2302 reviewed (approved):
Very good work, I like it
